File: frontend/src/app/features/projects/models/proyecto.models.ts
export interface ProyectoDto {
  id: number;
  nombre: string;
  descripcion: string | null;
  fechaInicio: string;
  fechaFin: string;
  estadoId: number;
  estadoNombre: string;
  creadoPorId: number;
  creadoPorNombre: string;
  fechaCreacion: string;
}

export interface PagedResultDto<T> {
  items: T[];
  total: number;
  pagina: number;
  tamanoPagina: number;
  totalPaginas: number;
}

export interface ApiResponse<T> {
  success: boolean;
  message?: string;
  data?: T;
}

export interface SaveProyectoPayload {
  nombre: string;
  descripcion: string | null;
  fechaInicio: string;
  fechaFin: string;
  estadoId: number;
}

export const ESTADOS_PROYECTO = [
  { id: 1, nombre: 'Pendiente' },
  { id: 2, nombre: 'En Progreso' },
  { id: 3, nombre: 'Completada' },
  { id: 4, nombre: 'Cancelada' }
] as const;

File: backend/Src/Core/Application/UseCases/Proyectos/ProyectoModels.cs
using Domain.Entities;

namespace Application.UseCases.Proyectos;

public record GetProyectosPagedRequest(int Pagina, int TamanoPagina);

public record GetProyectoByIdRequest(int Id);

public record CreateProyectoRequest(
    string Nombre,
    string? Descripcion,
    DateTime FechaInicio,
    DateTime FechaFin,
    int EstadoId);

public record UpdateProyectoRequest(
    int Id,
    string Nombre,
    string? Descripcion,
    DateTime FechaInicio,
    DateTime FechaFin,
    int EstadoId);

public record DeleteProyectoRequest(int Id);

public record ProyectoDto(
    int Id,
    string Nombre,
    string? Descripcion,
    DateTime FechaInicio,
    DateTime FechaFin,
    int EstadoId,
    string EstadoNombre,
    int CreadoPorId,
    string CreadoPorNombre,
    DateTime FechaCreacion)
{
    public static ProyectoDto FromEntity(Proyecto p) => new(
        p.Id,
        p.Nombre,
        p.Descripcion,
        p.FechaInicio,
        p.FechaFin,
        p.EstadoId,
        p.EstadoNombre,
        p.CreadoPorId,
        p.CreadoPorNombre,
        p.FechaCreacion);
}
